pkg/binary/envoy/fetch_test.go

```diff
@@ -50,17 +182,22 @@ func TestFetchIfNeeded(t *testing.T) {
 	defer manifestServer.Close()
 
 	t.Run("error on incorrect URL", func(t *testing.T) {
-		o.ManifestURL = manifestServer.URL + "/mannyfest.json"
+		out := new(bytes.Buffer)
+		o := &globals.GlobalOpts{HomeDir: homeDir, Out: out, ManifestURL: manifestServer.URL + "/mannyfest.json"}
 		_, e := FetchIfNeeded(o, r)
-		require.EqualError(t, e, "received 404 response code from "+o.ManifestURL)
+		require.EqualError(t, e, "received 404 status code from "+o.ManifestURL)
+		require.Empty(t, out)
 	})
 
-	o.ManifestURL = manifestServer.URL + "/manifest.json"
-
 	expectedPath := filepath.Join(homeDir, "builds", "standard", envoyVersion, "darwin", "bin", "envoy")
 	t.Run("downloads when doesn't exists", func(t *testing.T) {
+		out := new(bytes.Buffer)
+		o := &globals.GlobalOpts{HomeDir: homeDir, Out: out, ManifestURL: manifestServer.URL + "/manifest.json"}
 		envoyPath, e := FetchIfNeeded(o, r)
 		require.NoError(t, e)
+		require.Contains(t, out.String(), fmt.Sprintf("downloading %s/builds/1.17.2/darwin.tar.xz", manifestServer.URL))
```

Comment:
we tell users now what they are downloading which can help in debugging more than the progress bar would